cmd/fs-v1.go

```diff
@@ -731,14 +767,33 @@ func (fs fsObjects) ListObjects(bucket, prefix, marker, delimiter string, maxKey
 			objInfo.IsDir = true
 			return
 		}
+
+		objectLock := globalNSMutex.NewNSLock(bucket, entry)
+		objectLock.RLock()
+		var md5Sum string
+		md5Sum, err = fs.getObjectETag(bucket, entry)
```

Comment:
you can do RUnlock() here once.
Or move Lock/Unlock calls inside getObjectETag() function